Implement Azure Application Gateway create template deployment

milismsft opened the pull request:
Implement Azure Application Gateway create template deployment; unit test cases to test the implementation included as well.

Add subnet as part of the description for Spinnaker load balancer.

@JargoonPard @scotmoor PTAL

milismsft committed:
Implement Azure Application Gateway create template deployment

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy:
```diff
@@ -0,0 +1,64 @@
+  long capacity = 2
+
+  static class AzureAppGatewayHealthcheckProbe {
+    enum AzureLoadBalancerProbesType {
```
This should just be Protocol not AzureAppGatewayHealthcheckProbe and then can be reused in multiple places.

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy:
```diff
@@ -0,0 +1,64 @@
+  }
+
+  static class AzureAppGatewayRule {
+    enum AzureLoadBalancingRulesType {
```
Same comment as above on AzureAppGatewayHealthcheckProbe

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy:
```diff
@@ -0,0 +1,64 @@
+
+    String name
+    AzureLoadBalancerProbesType protocol = "HTTP"
+    String host = "localhost"
```
Did you confirm that localhost will resolve correctly?

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/appgateway/model/AzureAppGatewayDescription.groovy:
```diff
@@ -0,0 +1,64 @@
+  String securityGroup
+  String dnsName
+  String cluster
+  List<String> serverGroups
```
vnet to serverGroups, these values aren't ever passed in to an AppGateway for creation, why are they needed?

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/model/AzureLoadBalancer.groovy:
```diff
@@ -27,6 +27,7 @@ class AzureLoadBalancer implements LoadBalancer {
   String name
   String region
   String vnet
+  String subnet
```
We don't render this anywhere why was it added?

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/model/AzureLoadBalancerDescription.groovy:
```diff
@@ -21,6 +21,7 @@ import com.netflix.spinnaker.clouddriver.azure.resources.common.AzureResourceOps
 class AzureLoadBalancerDescription extends AzureResourceOpsDescription {
   String loadBalancerName
   String vnet
+  String subnet
```
We don't render this anywhere why was it added?

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerController.groovy:
```diff
@@ -73,6 +73,7 @@ class AzureLoadBalancerController {
       lbDetail.createdTime = azureLoadBalancerDescription.createdTime
       lbDetail.serverGroup = azureLoadBalancerDescription.serverGroup
       lbDetail.vnet = azureLoadBalancerDescription.vnet ?: "vnet-unassigned"
+      lbDetail.subnet = azureLoadBalancerDescription.subnet ?: "subnet-unassigned"
```
We don't render this anywhere why was it added?

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/loadbalancer/view/AzureLoadBalancerProvider.groovy:
```diff
@@ -78,6 +78,7 @@ class AzureLoadBalancerProvider implements LoadBalancerProvider<AzureLoadBalance
       name: loadBalancerDescription.loadBalancerName,
       region: loadBalancerDescription.region,
       vnet: loadBalancerDescription.vnet?: "vnet-unassigned",
+      subnet: loadBalancerDescription.subnet?: "subnet-unassigned",
```
We don't render this anywhere why was it added?

JargoonPard commented on clouddriver-azure/src/main/groovy/com/netflix/spinnaker/clouddriver/azure/resources/servergroup/ops/CreateAzureServerGroupAtomicOperation.groovy:
```diff
@@ -110,6 +110,7 @@ class CreateAzureServerGroupAtomicOperation implements AtomicOperation<Map> {
       lbDescription.detail = description.detail
       lbDescription.securityGroup = description.securityGroup?.name
       lbDescription.vnet = virtualNetworkName
+      lbDescription.subnet = subnetName
```
We don't render this anywhere why was it added?

JargoonPard commented:
LGTM

tomaslin committed:
add a bastionHost field in titus

Commit:
provider/kubernetes: correctly report job failure state

Commit:
core: Prefix job controller path with containing app name

Commit:
cats: Use redis rather than system for timestamp

lwander committed:
Merge pull request #555 from lwander/cats-redis-time

cats: Use redis rather than system for timestamp

tomaslin committed:
Merge pull request #551 from tomaslin/add-bastion-host-to-titus

Titus: add a bastionHost field

Commit:
provider/google: Avoid NPE when zone is not passed. (#557)

lwander committed:
Merge pull request #552 from lwander/kubernetes-job-result

provider/kubernetes: correctly report job failure state

lwander committed:
Merge pull request #553 from lwander/kubernetes-job-by-app

core: Prefix job controller path with containing app name

ajordens committed:
ReservationReportCachingAgent should treat reservations from a VPC-only account as VPC

- Also split terminates from the CleanupDetachedInstancesAgent into groups of 20 in order to avoid any Amazon constraints on # of concurrent terminations

tomaslin committed:
titus iamProfile support

tomaslin committed:
Merge pull request #560 from tomaslin/titus-iamprofile

titus iamProfile support

ajordens committed:
Treat non-existent credentials as a 404

ajordens committed:
Merge pull request #559 from ajordens/reservationReport-support-vpc-only-accounts

ReservationReportCachingAgent should treat reservations from a VPC-on…

ajordens committed:
AmazonClientProvider support for constructing AWSLambda clients

ajordens committed:
Merge pull request #561 from ajordens/amazon-lambda-client

AmazonClientProvider support for constructing AWSLambda clients

Commit:
provider/aws: The list() entrypoint in AmazonLoadBalancerController should not query for all load balancer identifiers. When other providers are caching in the same redis instance, NPEs are thrown. (#562)

provider/cf: The list() entrypoint in CloudFoundryLoadBalancerController should not query for all load balancer identifiers.
